caravel/assets/spec/javascripts/sqllab/reducers_spec.js

```diff
@@ -25,4 +26,73 @@ describe('sqlLabReducer', () => {
       expect(newState.tabHistory[1]).to.eq(newState.queryEditors[1].id);
     });
   });
+  describe('Alerts', () => {
+    const state = Object.assign({}, initialState);
+    let newState;
+    it('should add one alert', () => {
+      newState = r.sqlLabReducer(state, actions.addAlert(alert));
+      expect(newState.alerts).to.have.lengthOf(1);
+    });
+    it('should remove one alert', () => {
+      newState = r.sqlLabReducer(newState, actions.removeAlert(newState.alerts[0]));
+      expect(newState.alerts).to.have.lengthOf(0);
+    });
+  });
+  describe('Query editors actions', () => {
+    const state = Object.assign({}, initialState);
+    let newState;
+    const defaultQueryEditor = state.queryEditors[0];
+    const qe = Object.assign({}, defaultQueryEditor);
+    qe.id = '2ed2';
+    it('should add a query editor', () => {
+      newState = r.sqlLabReducer(state, actions.addQueryEditor(qe));
+      expect(newState.queryEditors).to.have.lengthOf(2);
+    });
+    it('should remove a query editor', () => {
+      newState = r.sqlLabReducer(newState, actions.removeQueryEditor(qe));
+      expect(newState.queryEditors).to.have.lengthOf(1);
+    });
+    it('should set q query editor active', () => {
+      newState = r.sqlLabReducer(state, actions.addQueryEditor(qe));
+      newState = r.sqlLabReducer(newState, actions.setActiveQueryEditor(defaultQueryEditor));
+      expect(newState.tabHistory[newState.tabHistory.length - 1]).equals(defaultQueryEditor.id);
+    });
+    it('should not fail while setting attributes', () => {
+      newState = r.sqlLabReducer(state, actions.queryEditorSetDb(qe, 1));
+      newState = r.sqlLabReducer(state, actions.queryEditorSetSchema(qe, 'schema'));
+      newState = r.sqlLabReducer(state, actions.queryEditorSetAutorun(qe, false));
+      newState = r.sqlLabReducer(state, actions.queryEditorSetTitle(qe, 'title'));
+      newState = r.sqlLabReducer(state, actions.queryEditorSetSql(qe, 'SELECT nothing'));
+    });
+  });
+  describe('Tables', () => {
+    const state = Object.assign({}, initialState);
+    let newState;
+    it('should add a table', () => {
```

Comment:
looks like this could be broken up into 4 tests for each action: `mergeTable`, `collapseTable`, `expandTable`, `removeTable`.